aten/src/ATen/core/tensor_type.cpp

```diff
@@ -173,21 +211,35 @@ VaryingShape<Stride> TensorType::computeStrideProps(
         }
       }
     }
+    // conveniently is_contiguous_strides/is_contiguous_strides only returns
+    // true when there's no memory overlap, so we only re-compute has_overlap
+    // in the last branch when both returns false
+    if (!tensor_contiguity) {
+      // trust tensor_contiguity and only computes overlap when it is not set
+      has_overlap = possible_cross_dimension_overlap(sizes, strides);
```

Comment:
you should be computing overlap only if checks for `contiguous` and `channels_last` in lines 172 and 177 weren't true

Reply:
That's what's happening since this line is inside the false block of line 180.

Reply:
Oh, right, sorry I missed that